### PR TITLE
Avoid putting symbols in global to fix incompatibility with Temporal Sandbox.

### PR DESCRIPTION
**Pull request by mikearnaldi**

### CODE REVIEW

File: packages/effect/src/GlobalValue.ts [hunk 1]
```diff
@@ -46,6 +42,12 @@ const globalStore = (globalThis as any)[globalStoreId] as Map<unknown, any>
  * @since 2.0.0
  */
 export const globalValue = <A>(id: unknown, compute: () => A): A => {
+  if (!globalStore) {
+    // @ts-expect-error
+    globalThis[globalStoreId] = new Map()
```
Comment (tim-smart) [inline packages/effect/src/GlobalValue.ts:47]:
```suggestion
    globalThis[globalStoreId] ??= new Map()
```

Not sure if we want to overwrite if it already exists.

**Comment by izakfilmalter** [comment]
This pr is lit. Thank you @mikearnaldi and @mjameswh for helping me and getting this resolved.

File: packages/effect/src/GlobalValue.ts (hunk 1 above)
Comment (mikearnaldi) [reply packages/effect/src/GlobalValue.ts:47]: how can it already exist if `globalStore` is undefined?

Comment (tim-smart) [reply packages/effect/src/GlobalValue.ts]: It would in the case of a duplicated module.

**Pull request merged**